packages/ember-views/lib/components/component.js

```diff
@@ -456,6 +456,182 @@ var Component = View.extend(TargetActionSupport, {
     @public
     @property positionalParams
   */
+
+  /**
+    `didInitAttrs` runs on the first render after a component was
+    created and passed attrs are guaranteed to be present.
+
+    The attributes will be available as this.get('attrName') or by
+    accessing the `attrs` property of the options argument.
+
+    ```app/components/my-collection.js
+    export default Ember.Component.extend({
+      content: null,
+
+      didInitAttrs: function(options) {
+        var content = this.get('content');
+        if (!content) {
+          this.set('content', []);
+        }
+      }
+    });
+    ```
+
+    @public
+    @method didInitAttrs
+    @param options {Object}
+    @since 1.13.0
+  */
+
+  /**
+    `didReceiveAttrs` runs on the first render after `didInitAttrs`,
+    and it also runs on subsequent re-renders, which is useful for
+    logic that is the same on all renders. It does not run when the
+    component has been re-rendered from the inside.
+
+    ```app/components/feedback-form.js
+    export default Ember.Component.extend({
+     didReceiveAttrs: function(options) {
+        this.set('title', 'Please provide us with your feedback');
+      }
+    });
+    ```
+
+    @public
+    @method didReceiveAttrs
+    @param options {Object}
+    @since 1.13.0
+  */
+
+
+  /**
+    `willRender` runs before the template is rendered. It runs when
+    the template is updated for any reason (both initial and
+    re-render, and regardless of whether the change was caused by an
+    attrs change or re-render)
+
+    ```app/components/todo-footer.js
+    export default Ember.Component.extend({
+      morethanOne: false,
+
+      willRender: function() {
+        if (this.get('items.length') > 1) {
+          this.set('morethanOne', true);
+        }
+      }
+    });
+    ```
+
+    @public
+    @method willRender
+    @since 1.13.0
+  */
+
+  /**
+   Called when the element of the component has been inserted into the DOM
+   or after the view was re-rendered. Override this function to do any set up
+   that requires an element in the document body.
+
+
+    `willRender` runs before the template is rendered. It runs when
```

Comment:
Isn't `didInsertElement` already documented?  Perhaps it is on Ember.View though, so it doesn't show up properly?